src/ansys/fluent/core/quantity/units_table.py

```diff
@@ -0,0 +1,272 @@
+    filter_unit_term()
+        Separate multiplier, base, and power from a unit term.
+    si_data()
+        Compute the SI unit string, SI multiplier, and SI offset from a unit string of any type.
+    si_conversion()
+        Perform SI conversion based on quantity type.
+    condense()
+        Condenses a unit string by collecting like-terms.
+    get_type()
+        Returns the type associated with a unit string.
+
+    Returns
+    -------
+    UnitsTable instance.
+    """
+
+    def __init__(self):
+        file_path = os.path.relpath(__file__)
+        file_dir = os.path.dirname(file_path)
+        qc_path = os.path.join(file_dir, "quantity_config.yaml")
+
+        with open(qc_path, "r") as qc_yaml:
+            qc_data = yaml.safe_load(qc_yaml)
+
+        self._api_quantity_map: dict = qc_data["api_quantity_map"]
+        self._fundamental_units: dict = qc_data["fundamental_units"]
+        self._derived_units: dict = qc_data["derived_units"]
+        self._multipliers: dict = qc_data["multipliers"]
+
+    def _has_multiplier(self, unit_term: str) -> bool:
+        """Check if a unit term contains a multiplier.
+
+        Parameters
+        ----------
+        unit_term : str
+            Unit term of a unit string.
+
+        Returns
+        -------
+        : bool
+            Boolean of multiplier within unit_term.
+        """
+        # Check if the unit term is not an existing fundamental or derived unit.
+        return not (
+            (unit_term in self._fundamental_units) or (unit_term in self._derived_units)
+        )
+
+    def _si_map(self, unit_term: str) -> str:
+        """Maps unit to SI unit equivalent.
+
+        Parameters
+        ----------
+        unit_term : str
+            Unit term of a unit string.
+
+        Returns
+        -------
+        term : str
+            SI unit equivalent.
+        """
+        # Retrieve type associated with unit term
+        unit_term_type = self._fundamental_units[unit_term]["type"]
+
+        # Find SI unit with same type as unit term
+        for term, term_info in self._fundamental_units.items():
+            if term_info["type"] == unit_term_type and term_info["factor"] == 1.0:
+                return term
+
+    @property
+    def api_quantity_map(self):
+        """Quantity map values from settings API."""
+        return self._api_quantity_map
+
+    @property
+    def fundamental_units(self):
+        """Fundamental units and properties representing Mass, Length, Time, Current, Chemical Amount, Light, Solid Angle, Angle, Temperature and Temperature Difference."""
+        return self._fundamental_units
+
+    @property
+    def derived_units(self):
+        """Derived units and properties composed of fundamental units."""
+        return self._derived_units
+
+    @property
+    def multipliers(self):
+        """Multiplier prefixes and respective factors."""
+        return self._multipliers
+
+    def filter_unit_term(self, unit_term: str) -> tuple:
```

Comment:
It would be good to directly unit-test this code to ensure that it is always returning the three components correctly.

Reply:
Added unit tests for `filter_unit_term`: 

https://github.com/ansys/pyfluent/blob/933480e02ff41edef549668b8a33b66019215349/tests/test_quantity.py#L961-L968